server/jetty/src/main/java/io/deephaven/server/jetty/JettyConfig.java

```diff
@@ -68,15 +77,60 @@ public int port() {
     }
 
     /**
-     * Include websockets. Defaults to {@value DEFAULT_WITH_WEBSOCKETS}.
+     * Include websockets.
      */
-    @Default
-    public boolean websockets() {
-        return DEFAULT_WITH_WEBSOCKETS;
+    @Nullable
+    public abstract Boolean websockets();
+
+    /**
+     * Include HTTP/1.1.
+     */
+    @Nullable
+    public abstract Boolean http1();
+
+    /**
+     * Returns {@link #websockets()} if explicitly set. If {@link #proxyHint()} is {@code true}, returns {@code false}.
+     * Otherwise, defaults to {@code true} when {@link #ssl()} is empty, and {@code false} when {@link #ssl()} is
+     * present.
+     */
+    public final boolean websocketsOrDefault() {
+        final Boolean websockets = websockets();
+        if (websockets != null) {
+            return websockets;
+        }
+        if (Boolean.TRUE.equals(proxyHint())) {
+            return false;
+        }
+        return ssl().isEmpty();
+    }
+
+    /**
+     * Returns {@link #http1()} if explicitly set. If {@link #proxyHint()} is {@code true}, returns {@code false}.
+     * Otherwise, defaults to {@code true} when {@link #ssl()} is empty, and {@code false} when {@link #ssl()} is
+     * present.
+     */
+    public final boolean http1OrDefault() {
+        final Boolean http1 = http1();
+        if (http1 != null) {
+            return http1;
+        }
+        if (Boolean.TRUE.equals(proxyHint())) {
+            return false;
+        }
+        return ssl().isEmpty();
```

Comment:
if ssl is enabled without a proxy, have you confirmed that a browser can connect to it without needing to first try http/1.1? It has been a while since I tested that, and recall some hiccup there.

Reply:
At least with my version of Firefox on Linux, yes. I can't speak for Chrome, older versions, and other operating systems at this point in time. It would surprise me a bit if anything relatively modern couldn't handle it - but might be worth some further testing?

Reply:
I'd vote do the testing before turning it off, yes - bad form to ship it and then find out we broke it. 

Alternatively, just be less aggressive about disabling http/1.1, and merge the rest.

Reply:
I've set it to be more conservative, and filed a follow-up issue.